behaviour/graql/reasoner/resolution/value-predicate.feature

```diff
@@ -0,0 +1,901 @@
+#
+# Copyright (C) 2020 Grakn Labs
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+Feature: Value Predicate Resolution
+
+  Background: Set up keyspaces for resolution testing
+
+    Given connection has been opened
+    Given connection delete all keyspaces
+    Given connection open sessions for keyspaces:
+      | materialised |
+      | reasoned     |
+    Given materialised keyspace is named: materialised
+    Given reasoned keyspace is named: reasoned
+    Given for each session, graql define
+      """
+      define
+
+      person sub entity,
+        plays leader,
+        plays team-member,
+        has string-attribute,
+        has unrelated-attribute,
+        has sub-string-attribute,
+        has name,
+        has age,
+        has is-old,
+        key ref;
+
+      tortoise sub entity,
+        has age,
+        has is-old,
+        key ref;
+
+      soft-drink sub entity,
+        has name,
+        has retailer,
+        key ref;
+
+      team sub relation,
+        relates leader,
+        relates team-member,
+        has string-attribute,
+        key ref;
+
+      string-attribute sub attribute, value string;
+      retailer sub attribute, value string;
+      age sub attribute, value long;
+      name sub attribute, value string;
+      is-old sub attribute, value boolean;
+      sub-string-attribute sub string-attribute;
+      unrelated-attribute sub attribute, value string;
+      ref sub attribute, value long;
+      """
+
+
+  # TODO: change 'then' block to "$x has is-old true" once implicit attribute variables are resolvable
+  Scenario: a rule can infer an attribute ownership based on a value predicate
+    Given for each session, graql define
+      """
+      define
+      tortoises-become-old-at-age-1-year sub rule,
+      when {
+        $x isa tortoise, has age $a;
+        $a > 0;
+      },
+      then {
+        $x has is-old $t;
```

Comment:
Is the `$t` here a concession made because of the relevant resolution test framework limitation

Reply:
Yes - and it says so in a TODO.